### PR TITLE
Fix EnvoyFilter for Istio 1.7

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**
Updated EnvoyFilter syntax for Istio 1.7 with fallback to old version in old environments running on Istio 1.4

Changes proposed in this pull request:

- Updated EnvoyFilter syntax for Istio 1.7

**Related issue(s)**
See also: https://github.com/kyma-project/kyma/pull/9503/files

### CODE REVIEW

File: chart/compass/templates/rewrite-client-cert-filter.yaml [hunk 1]
```diff
@@ -21,7 +21,8 @@ spec:
         operation: INSERT_BEFORE
         value:
           name: envoy.lua
-          config:
+          typed_config:
```
Comment (piotrmsc) [inline chart/compass/templates/rewrite-client-cert-filter.yaml:24]: it won't simply work this way. Compass repo is still running on kyma version with istio 1.4.

File: chart/compass/templates/rewrite-token-filter.yaml [hunk 2]
```diff
@@ -21,7 +21,8 @@ spec:
         operation: INSERT_BEFORE
         value:
           name: envoy.lua
-          config:
+          typed_config:
+            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
```
Comment (Demonsthere) [inline chart/compass/templates/rewrite-token-filter.yaml:25]:
```suggestion
         {{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1/PeerAuthentication" }}
          typed_config:
            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
          {{- end }}
```

File: chart/compass/templates/rewrite-client-cert-filter.yaml [hunk 3]
```diff
@@ -21,7 +21,8 @@ spec:
...
+            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
```
Comment (Demonsthere) [inline chart/compass/templates/rewrite-client-cert-filter.yaml:25]:
```suggestion
          {{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1/PeerAuthentication" }}
          typed_config:
            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
          {{- end }} 
```

File: chart/compass/templates/rewrite-token-filter.yaml (hunk 2 above)
Comment (Demonsthere) [inline chart/compass/templates/rewrite-token-filter.yaml:25]:
```suggestion
          {{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1/PeerAuthentication" }}
          typed_config:
            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
          {{- else }}
          config:
          {{- end }} 
```

File: chart/compass/templates/rewrite-client-cert-filter.yaml (hunk 3 above)
Comment (Demonsthere) [inline chart/compass/templates/rewrite-client-cert-filter.yaml:25]:
```suggestion
          {{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1/PeerAuthentication" }}
          typed_config:
            "@type": "type.googleapis.com/envoy.config.filter.http.lua.v2.Lua"
          {{- else }}
          config:
          {{- end }} 
```

**Review by Demonsthere — CHANGES_REQUESTED** [review]

**Review by Demonsthere — APPROVED** [review]

**Review by rafalpotempa — APPROVED** [review]

**Pull request merged**